### PR TITLE
DPE-1512 Autogenerated username+database when config values empty for legacy relations

### PR DESCRIPTION
**Pull request by shayancanonical**
## Issue
- We would like to auto-generate the username and database for the legacy mysql relations if the config values for them are empty.
- There is a divergence in logic between the `mysql` and `mysql-root` legacy relation implementations. Furthermore, some of the files and methods reference `osm_mysql` (instead of `mysql_root`)

## Solution
- Auto-generate username and database if config values for these options are empty (port over logic from the vm operator)
- Add missing logic in both `mysql` and `mysql_root` to make their logic as similar as possible. Rename `osm_mysql` to `mysql_root`

### CODE REVIEW

**Review by paulomach — DISMISSED** [review]
small nit, otherwise lgtm

File: src/relations/mysql.py [hunk 1]
```diff
@@ -141,7 +172,7 @@ def _on_peer_relation_changed(self, event: RelationChangedEvent) -> None:
             event.defer()
             return
 
-        if (relation_data := self.charm.app_peer_data.get("mysql_relation_data", "{}")) == "{}":
+        if (relation_data := self.charm.app_peer_data.get(MYSQL_RELATION_DATA_KEY, "{}")) == "{}":
```
[the remark this answers is not in the document]
Comment (shayancanonical) [reply src/relations/mysql.py]: addressed in 52141cb5a564d653787824fea1a0b4bf8511722e

File: src/relations/mysql.py [hunk 2]
```diff
@@ -78,30 +125,14 @@ def _on_leader_elected(self, _) -> None:
...
-
     def _update_status(self, _) -> None:
         """Handle the update status event.
 
         Compares the current host (in relation data) with the current primary.
         If they are different, changes a key in the peer relation databag in order
         to trigger the peer relation changed event.
         """
-        if (relation_data := self.charm.app_peer_data.get("mysql_relation_data", "{}")) == "{}":
+        if (relation_data := self.charm.app_peer_data.get(MYSQL_RELATION_DATA_KEY, "{}")) == "{}":
```
[the remark this answers is not in the document]
Comment (shayancanonical) [reply src/relations/mysql.py]: addressed in 52141cb5a564d653787824fea1a0b4bf8511722e

**Review by paulomach — APPROVED** [review]

**Review by taurus-forever — APPROVED** [review]
Thank you!

**Pull request merged**